src/journal/compress.c

```diff
@@ -49,15 +55,38 @@ DEFINE_TRIVIAL_CLEANUP_FUNC(LZ4F_compressionContext_t, LZ4F_freeCompressionConte
 DEFINE_TRIVIAL_CLEANUP_FUNC(LZ4F_decompressionContext_t, LZ4F_freeDecompressionContext);
 #endif
 
+#if HAVE_ZSTD
+DEFINE_TRIVIAL_CLEANUP_FUNC(ZSTD_CCtx*, ZSTD_freeCCtx);
+DEFINE_TRIVIAL_CLEANUP_FUNC(ZSTD_DCtx*, ZSTD_freeDCtx);
+#endif
+
 #define ALIGN_8(l) ALIGN_TO(l, sizeof(size_t))
 
 static const char* const object_compressed_table[_OBJECT_COMPRESSED_MAX] = {
         [OBJECT_COMPRESSED_XZ] = "XZ",
         [OBJECT_COMPRESSED_LZ4] = "LZ4",
+        [OBJECT_COMPRESSED_ZSTD] = "ZSTD",
 };
 
 DEFINE_STRING_TABLE_LOOKUP(object_compressed, int);
 
+#if HAVE_ZSTD
+static int zstd_ret_to_errno(size_t ret) {
+        static_assert(ZSTD_VERSION_NUMBER >= 10301, "<zstd_errors.h> is stable since zstd 1.3.1");
```

Comment:
We usually use `assert_cc()` instead of `static_assert()`. When you use `assert_cc()` here, then it is not necessary to include assert.h. Just a comment.